spectral_cube/lower_dimensional_structures.py

```diff
@@ -1058,23 +1082,26 @@ def _new_spectrum_with(self, data=None, wcs=None, mask=None, meta=None,
         fill_value = self._fill_value if fill_value is None else fill_value
         spectral_unit = self._spectral_unit if spectral_unit is None else u.Unit(spectral_unit)
 
-        if beams is None:
-            if hasattr(self, 'beams'):
-                kwargs['beams'] = beams
-        elif beam is None:
-            if hasattr(self, 'beam'):
+        if beams is None and not hasattr(self, 'beams'):
```

Comment:
@keflavich can you see a cleaner way to do this? Basically, I want to avoid checking for `self.beam` when it's a VRODS.

Reply:
This would be solved by inheriting from a `BaseOneDSpectrum`, too